tensorflow/lite/python/lite_v2_test.py

```diff
@@ -1192,5 +1192,81 @@ def model(in_tensor):
         str(error.exception))
 
 
+class AffineOpThenMulFusionTest(lite_v2_test_util.ModelTest):
+
+  @test_util.run_v2_only
+  def testAffineOpThenFusion(self):
+    # Test fusion of (x @ y) * z into fullyconnected
+    input_data1 = tf.constant([1., 2.], shape=[1, 2])
+
+    @tf.function
+    def func1(x):
+      y_const = tf.constant([3., 4., 5., 6.], shape=[2, 2])
+      z_const = tf.constant([7., 8.], shape=[1, 2])
+      return tf.matmul(x, y_const) * z_const
+
+    @tf.function
+    def func2(x):
+      y_const = tf.constant([3., 4., 5., 6.], shape=[2, 2])
+      z_const = tf.constant([7., 8., 9., 10.], shape=[2, 2])
+      return tf.matmul(x, y_const) * z_const
+
+    @tf.function
+    def func3(x):
+      y_const = tf.constant([3., 4., 5., 6.], shape=[2, 2])
+      z_const = tf.constant([7., 8.], shape=[2, 1])
+      return tf.matmul(x, y_const) * z_const
+
+    # Test fusion of (x ∗ y) * z into conv2d
+    input_data2 = tf.constant([1., 2.], shape=[1, 1, 2, 1])
+
+    @tf.function
+    def func4(x):
+      y_const = tf.constant([3., 4., 5., 6.], shape=[2, 1, 1, 2])
+      z_const = tf.constant([7., 8.], shape=[2, 1])
+      return tf.nn.conv2d(x, y_const, strides=[1, 1, 1, 1],
+                          padding='SAME') * z_const
+
+    @tf.function
+    def func5(x):
+      y_const = tf.constant([3., 4., 5., 6.], shape=[2, 1, 1, 2])
+      z_const = tf.constant([7., 8.], shape=[1, 2])
+      return tf.nn.conv2d(x, y_const, strides=[1, 1, 1, 1],
+                          padding='SAME') * z_const
+
+    test_cases = [
+        (input_data1, func1, 1), (input_data1, func2, 2),
+        (input_data1, func3, 2), (input_data2, func4, 2),
+        (input_data2, func5, 1)]
+
+    for case in test_cases:
+      input_data, func, expected_number_of_ops = case
+      root = tracking.AutoTrackable()
+      root.f = func
+      concrete_func = root.f.get_concrete_function(input_data)
+
+      # Convert model.
+      converter = lite.TFLiteConverterV2.from_concrete_functions(
+          [concrete_func])
+      tflite_model = converter.convert()
+
+      # Check values from converted model.
+      expected_value = root.f(input_data)
+      converter.experimental_new_converter = True
+
+      actual_value = self._evaluateTFLiteModel(tflite_model, [input_data])
+      np.testing.assert_almost_equal(expected_value.numpy(), actual_value[0])
+
+      # Enable hybrid quantization, same result
+      converter.experimental_new_converter = True
```

Comment:
duplicate and not needed. Remove please

Reply:
Good catch, this code is still from the time when it wasn't enabled by default.